contrib/epee/include/net/http_protocol_handler.inl

```diff
@@ -345,7 +345,12 @@ namespace net_utils
 		{
 			analize_http_method(result, m_query_info.m_http_method, m_query_info.m_http_ver_hi, m_query_info.m_http_ver_hi);
 			m_query_info.m_URI = result[10];
-      parse_uri(m_query_info.m_URI, m_query_info.m_uri_content);
+			if (!parse_uri(m_query_info.m_URI, m_query_info.m_uri_content))
+			{
+				m_state = http_state_error;
```

Comment:
Not sure where the dev team stands on macros w/ return statements (in this case it could also just be another function), but this pattern is used enough in this file it could be cleaned up with something like FAIL_CONNECTION(<reason>) that handles the m_state state and appropriate logging.

Reply:
It's a bit subjective, and depends on the case. In general, I try not to change code gratuitously.